client/src/components/layouts/default.tsx

```diff
@@ -237,6 +248,49 @@ function DefaultLayout({
   }
 }
 
+const useGetAllBlockIds = () => {
```

Comment:
```suggestion
// TODO: get challenge nodes directly rather than wrapped in edges
const useGetAllBlockIds = () => {
```
It's a little out of scope, so I think a TODO is enough.